server/container/cassandra-guice/src/main/java/org/apache/james/jmap/JMAPModule.java

```diff
@@ -42,14 +45,23 @@ protected void configure() {
 
     @Provides
     @Singleton
-    JMAPConfiguration provideConfiguration(FileSystem fileSystem) throws FileNotFoundException, ConfigurationException{
+    JMAPConfiguration provideConfiguration(FileSystem fileSystem) throws ConfigurationException, IOException{
         PropertiesConfiguration configuration = getConfiguration(fileSystem);
-        String keystore = configuration.getString("tls.keystoreURL");
-        String secret = configuration.getString("tls.secret");
-        return new JMAPConfiguration(keystore, secret);
+        return JMAPConfiguration.builder()
+                .keystore(configuration.getString("tls.keystoreURL"))
+                .secret(configuration.getString("tls.secret"))
+                .jwtPublicKeyPem(loadPublicKey(fileSystem, Optional.ofNullable(configuration.getString("jwt.publickeypem.url"))))
+                .build();
     }
 
     private PropertiesConfiguration getConfiguration(FileSystem fileSystem) throws FileNotFoundException, ConfigurationException {
         return new PropertiesConfiguration(fileSystem.getFile(FileSystem.FILE_PROTOCOL_AND_CONF + "jmap.properties"));
     }
+
+    private String loadPublicKey(FileSystem fileSystem, Optional<String> jwtPublickeyPemUrl) throws FileNotFoundException, IOException {
```

Comment:
Sorry but it doesn't make sense : you have an optional input url but a null string as an output ...